rom/dev/README.md

```diff
@@ -242,50 +254,85 @@ Both UDS and Field Entropy are available only during cold reset of Caliptra.
 
 | Slot | Key Vault | PCR Bank | Data Vault 48 Byte (Sticky) | Data Vault 4 Byte (Sticky) |
 |------|-----------|----------|-----------------------------|----------------------------|
-| 0 | UDS (48 bytes) | | | |
-| 1 | Field Entropy (32 bytes) | | | |
+| 0 | UDS for ECDSA (48 bytes) | | | |
+| 1 | Field Entropy for ECDSA (32 bytes) | | | |
+| 10 | Conditioned UDS for MLDSA | | | |
```

Comment:
I don't think we need to allocate extra KV slots for MLDSA-specific seeds. Every time we use a seed to generate an ECDSA key, we first condition it with a KDF and a context string. For example see [here](https://github.com/chipsalliance/caliptra-sw/blob/f102017b28c27642c50ad04a748d13e05dc35655/rom/dev/src/flow/cold_reset/idev_id.rs#L189) for IDevID generation. The conditioned values are stored in `KEY_ID_TMP` and then used to derive the asymmetric key.

We can simply pass a different context string when conditioning the UDS/CDIs for MLDSA generation.

Reply:
We do pass a different string when creating the seed for MLDSA key generation as noted below:

"Derive the ECDSA CDI using ROM specified label and UDS in Key Vault Slot 0 as data and store the resultant MAC in Key Vault Slot 6

hmac384_kdf(KvSlot0, b"idevid_cdi", KvSlot6)

If Dual-Cert Mode is enabled, use the conditioned UDS to derive the MLDSA CDI and store the resultant MAC in Key Vault Slot 12

hmac512_kdf(KvSlot10, b"idevid_mldsa_cdi", KvSlot12)"

Have a reconditioned UDS/FE was something that was discussed and agreed upon on the FW sync. Do you think using the same UDS/FE and just having separate strings for CDI (seed) generation (via KDF) is sufficient from a security standpoint?

Reply:
> Have a reconditioned UDS/FE was something that was discussed and agreed upon on the FW sync.

Ah sorry I wasn't around for that discussion.

> Do you think using the same UDS/FE and just having separate strings for CDI (seed) generation (via KDF) is sufficient from a security standpoint?

Yes. The CDI tree is algorithm-agnostic; it's just a sequence of secret seeds. Before any such seed is used for any purpose such as ECC keygen, the seed is run through a KDF. So we can keep the CDI tree the same and just tweak the KDF labels used at the leaves when creating asymmetric keys.

Reply:
Ok, I'll address this as a separate update to the document.